io/src/main/scala/fs2/io/net/SocketGroup.scala

```diff
@@ -94,11 +94,13 @@ private[net] object SocketGroup {
         options: List[SocketOption]
     ): Resource[F, Socket[F]] = {
       def setup: F[AsynchronousSocketChannel] =
-        Async[F].delay {
+        Async[F].defer {
           val ch =
             AsynchronousChannelProvider.provider.openAsynchronousSocketChannel(channelGroup)
-          options.foreach(opt => ch.setOption(opt.key, opt.value))
-          ch
+          Either
+            .catchNonFatal(options.foreach(opt => ch.setOption(opt.key, opt.value)))
```

Comment:
Hm, why is this `catchNonFatal` necessary? The old code would still result in any exception getting thrown from `ch.setOption` failing the overall effect.